pkg/registry/node/strategy.go

```diff
@@ -153,17 +153,30 @@ func ResourceLocation(getter ResourceGetter, connection client.ConnectionInfoGet
 	}
 	host := hostIP.String()
 
-	if portReq == "" || strconv.Itoa(ports.KubeletPort) == portReq {
-		// Ignore requested scheme, use scheme provided by GetConnectionInfo
+	// We check if we want to get a default Kubelet's transport. It happens if either:
+	// - no port is specified in request (Kubelet's port is default),
+	// - we're using Port stored as a DaemonEndpoint and requested port is a Kubelet's port stored in the DaemonEndpoint,
+	// - there's no information in the API about DaemonEnpoint (legacy cluster) and requested port is equal to ports.KubeletPort (cluster-wide config)
+	if portReq == "" ||
+		(node.Status.DaemonEndpoints.KubeletEndpoint.Port != 0 &&
+			strconv.Itoa(node.Status.DaemonEndpoints.KubeletEndpoint.Port) == portReq) ||
+		(node.Status.DaemonEndpoints.KubeletEndpoint.Port == 0 &&
+			strconv.Itoa(ports.KubeletPort) == portReq) {
 		scheme, port, kubeletTransport, err := connection.GetConnectionInfo(host)
 		if err != nil {
 			return nil, nil, err
 		}
+		var portString string
+		if node.Status.DaemonEndpoints.KubeletEndpoint.Port != 0 {
+			portString = strconv.Itoa(node.Status.DaemonEndpoints.KubeletEndpoint.Port)
+		} else {
+			portString = strconv.FormatUint(uint64(port), 10)
```

Comment:
oh, gotcha. That makes more sense now, sorry!

On Wed, Oct 14, 2015 at 3:17 AM, Marek Grabowski <notifications@github.com>
wrote:

> In pkg/registry/node/strategy.go
> <https://github.com/kubernetes/kubernetes/pull/12919#discussion_r41976630>
> :
>
> >  		scheme, port, kubeletTransport, err := connection.GetConnectionInfo(host)
> >  		if err != nil {
> >  			return nil, nil, err
> >  		}
> > +		var portString string
> > +		if node.Status.DaemonEndpoints.KubeletEndpoint.Port != 0 {
> > +			portString = strconv.Itoa(node.Status.DaemonEndpoints.KubeletEndpoint.Port)
> > +		} else {
> > +			portString = strconv.FormatUint(uint64(port), 10)
>
> OK, now I remember - port is an uint and there's not FormatUint for
> uint32 type. golang FTW!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/12919/files#r41976630>.
>